lib/livebook_web/live/session_live/secrets_component.ex

```diff
@@ -282,7 +282,13 @@ defmodule LivebookWeb.SessionLive.SecretsComponent do
   end
 
   def handle_event("grant_access", %{"secret_name" => secret_name}, socket) do
-    grant_access(socket.assigns.saved_secrets, secret_name, :app, socket)
+    cond do
+      app?(socket, secret_name) ->
+        grant_access(socket.assigns.saved_secrets, secret_name, :app, socket)
+
+      hub?(socket, secret_name) ->
+        grant_access(socket.assigns.saved_secrets, secret_name, :hub, socket)
+    end
```

Comment:
The idea here is to do something like: 

```
def handle_event("grant_access", %{"secret_name" => secret_name} = params, socket) do
  grant_access(socket.assigns.saved_secrets, secret_name, build_origin(params), socket)
  ...
```

Then we will have access to the `hub_id` inside `grant_access`.

We should refactor `select_secret` in a very similar way.

Reply:
In this case, we don't know the secret origin, we only have the name. When only calls this event when receive the `missing_secret` event from the `System.fetch_env!("LB_" <> _)`.